ui/server/src/main/scala/pl/touk/nussknacker/ui/security/AuthenticationConfigurationFactory.scala

```diff
@@ -0,0 +1,85 @@
+package pl.touk.nussknacker.ui.security
+
+import java.io.File
+import java.net.URI
+
+import com.typesafe.config.{Config, ConfigFactory}
+import com.typesafe.scalalogging.LazyLogging
+import net.ceedubs.ficus.readers.ValueReader
+import pl.touk.nussknacker.ui.security.AuthenticationConfigurationFactory.DefaultConfigUser
+import pl.touk.nussknacker.ui.security.api.GlobalPermission.GlobalPermission
+import pl.touk.nussknacker.ui.security.api.Permission.Permission
+import pl.touk.nussknacker.ui.security.basicauth.BasicAuthConfiguration
+import pl.touk.nussknacker.ui.security.oauth2.OAuth2Configuration
+
+trait AuthenticationConfiguration {
+  def authorizeUrl: Option[URI] = Option.empty
+  def backend: AuthenticationBackend.Value
+}
+
+case class DefaultAuthenticationConfiguration(backend: AuthenticationBackend.Value = AuthenticationBackend.Other, usersFile: String)
+  extends AuthenticationConfiguration with LazyLogging {
+  import net.ceedubs.ficus.Ficus._
+  import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+  import net.ceedubs.ficus.readers.EnumerationReader._
+
+  def loadUsers(): List[DefaultConfigUser]
+    = loadUsersConfig().as[List[DefaultConfigUser]](AuthenticationConfigurationFactory.usersConfigurationPath)
+
+  def loadUsersConfig(): Config
+    = ConfigFactory.parseFile(new File(usersFile))
+}
+
+//TODO: Add some connections between AuthenticatorProvider and AuthenticationConfigurationFactory
+object AuthenticationConfigurationFactory extends LazyLogging {
+  import net.ceedubs.ficus.Ficus._
+  import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+  import net.ceedubs.ficus.readers.EnumerationReader._
+
+  val usersConfigurationPath = "users"
+  val rulesConfigurationPath = "rules"
+
+  implicit val uriValueReader: ValueReader[URI] = new ValueReader[URI] {
+    def read(config: Config, path: String): URI = new URI(config.getString(path))
+  }
+
+  final val authenticationConfigPath = "authentication"
```

Comment:
`final` is redundant - val is already not modificable, final is to avoid override sth. And you can't extends objects ...